### PR TITLE
chore: add Konnect SDK factory

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

Add Konnect SDK factory.

### CODE REVIEW

File: controller/konnect/sdkfactory.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+type SDKToken string
+
+// SDKFactory is a factory for creating Konnect SDKs.
+type SDKFactory interface {
+	NewKonnectSDK(serverURL string, token SDKToken) *sdkkonnectgo.SDK
+}
+
+type sdkFactory struct {
+	ServerURL string
+	Token     SDKToken
```
Comment (mlavacca) [inline controller/konnect/sdkfactory.go]: What's the point of having these fields in the factory, since they are passed to the `NewKonnectSDK` function?

Comment (pmalek) [reply controller/konnect/sdkfactory.go]: Whoops, you're right.

File: controller/konnect/sdkfactory.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+}
```
Comment (pmalek) [inline controller/konnect/sdkfactory.go]:
```suggestion
type sdkFactory struct {}
```

File: controller/konnect/sdkfactory.go (hunk 1 above)
Comment (pmalek) [reply controller/konnect/sdkfactory.go]: Removed in 6466abce2253ee90f78c945d96d82ca53dd05745.

**Review by mlavacca — APPROVED** [review]

**Pull request merged**